edgedb-tokio/src/builder.rs

```diff
@@ -247,29 +249,30 @@ fn stash_path(project_dir: &Path) -> Result<PathBuf, Error> {
     Ok(config_dir()?.join("projects").join(stash_name(project_dir)))
 }
 
-fn is_valid_instance_name(name: &str) -> bool {
-    let mut chars = name.chars();
-    match chars.next() {
-        Some(c) if c.is_ascii_alphabetic() || c == '_' => {}
-        _ => return false,
-    }
-    for c in chars {
-        if !c.is_ascii_alphanumeric() && c != '_' {
-            return false;
-        }
-    }
-    return true;
-}
-
-fn is_valid_org_name(name: &str) -> bool {
+fn is_valid_instance_name(name: &str, is_cloud: bool) -> bool {
```

Comment:
Let's make two functions. Because:
1. For cloud it's not "instance name" it may also be org name
2. I also have troubles understanding what's going on with those conditions

So two functions should make it clearer, and please also copy the regexp that each of those matches to the docstring of the respective function.